### PR TITLE
Add Filesystem package

### PR DESCRIPTION
**Pull request by yrezgui**
- [x] Add filesystem demo
- [x] Upgrade sample project dependencies
- [x] Add POM details for `filesystem` package
- [x] Enable KDOC for `filesystem` package
- [x] Figure out how to avoid `minSdk` to **API 26** in the `build.gradle`
- [x] Fix API docs output directory 

@nic0lette Actual changes since your PR got merged: https://github.com/google/modernstorage/pull/35/files/3b7106c574606bd97f32eb777a06f0ddc49e0fb9..a13cd3a030e8d4c7fc8d80c18a842f5da0fd4254

### CODE REVIEW

File: filesystem/build.gradle [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    id 'org.jetbrains.dokka'
+    id 'com.vanniktech.maven.publish'
+}
+
+android {
+    compileSdkVersion 30
+    buildToolsVersion "30.0.3"
+
+    defaultConfig {
+        minSdkVersion 26
```
Comment (yrezgui) [inline filesystem/build.gradle:28]: @nic0lette I don't know where I should set the check for the `minSdk` to not trigger a compilation error. Any idea?

File: sample/src/main/java/com/google/modernstorage/sample/filesystem/FileSystemFragment.kt [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            actionOpenImageFile.launch(arrayOf("image/png", "image/jpg", "image/webp"))
+        }
+    }
+
+    private val actionOpenTextFile = registerForActivityResult(OpenDocument()) { uri ->
+        viewModel.onFileSelect(uri, FileType.TEXT)
+    }
+
+    private val actionOpenImageFile = registerForActivityResult(OpenDocument()) { uri ->
+        viewModel.onFileSelect(uri, FileType.IMAGE)
```
Comment (nic0lette) [inline sample/src/main/java/com/google/modernstorage/sample/filesystem/FileSystemFragment.kt:97]: `uri` is actually nullable -- not picking a file (pressing back out of DocsUI) causes this to crash.

File: sample/src/main/java/com/google/modernstorage/sample/filesystem/FileSystemFragment.kt [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.activity.result.contract.ActivityResultContracts.OpenDocument
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import com.google.android.material.snackbar.Snackbar
+import com.google.modernstorage.sample.R
+import com.google.modernstorage.sample.databinding.FragmentFilesystemBinding
+
+class FileSystemFragment : Fragment() {
```
Comment (nic0lette) [inline sample/src/main/java/com/google/modernstorage/sample/filesystem/FileSystemFragment.kt:32]: Comment about this class (and probably calling out that it's only 26+, along with a `@RequiresApi(Build.VERSION_CODES.O)` annotation.

File: sample/src/main/java/com/google/modernstorage/sample/DemoListAdapter.kt [hunk 4]
```diff
@@ -13,40 +13,45 @@
...
+import androidx.annotation.DrawableRes
 import androidx.annotation.IdRes
 import androidx.annotation.StringRes
 import androidx.navigation.findNavController
 import androidx.recyclerview.widget.RecyclerView
-import com.google.modernstorage.sample.R
+import com.google.modernstorage.sample.databinding.ListRowItemBinding
 
-data class Demo(@StringRes val nameRes: Int, @IdRes val actionRes: Int)
+data class Demo(
```
Comment (nic0lette) [inline sample/src/main/java/com/google/modernstorage/sample/DemoListAdapter.kt:27]: Probably best to put a comment on this (and the other classes) in this file.

File: sample/src/main/java/com/google/modernstorage/sample/filesystem/FileSystemViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import kotlinx.parcelize.Parcelize
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.stream.Collectors
+
+const val CURRENT_DOCUMENT_KEY = "currentDocument"
+
+class FileSystemViewModel(
```
Comment (nic0lette) [inline sample/src/main/java/com/google/modernstorage/sample/filesystem/FileSystemViewModel.kt:41]: Comment about this class (and probably calling out that it's only 26+, along with a @RequiresApi(Build.VERSION_CODES.O) annotation.

**Review by nic0lette — COMMENTED** [review]
First pass is mostly good. Thinking about the `minSdkVersion` thing :)

File: sample/src/main/java/com/google/modernstorage/sample/filesystem/FileSystemViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                    FileType.IMAGE -> loadImageContent(path)
+                }
+            }
+        }
+    }
+
+    @Suppress("BlockingMethodInNonBlockingContext")
+    private suspend fun loadTextContent(path: Path) {
+        withContext(Dispatchers.IO) {
+            val reader = Files.newBufferedReader(path)
```
Comment (nic0lette) [inline sample/src/main/java/com/google/modernstorage/sample/filesystem/FileSystemViewModel.kt:78]: You can also use `Files.readAllLines(path)` to just get all the lines ;)

**Review by nic0lette — APPROVED** [review]

**Pull request merged**